File: CommonBusiness/src/main/java/com/company/commonbusiness/base/activity/BaseMvpActivity.java
package com.company.commonbusiness.base.activity;

import com.company.commonbusiness.base.mvp.BasePresenter;
import com.company.commonbusiness.base.mvp.IMvpView;

/**
 * @author 李昭鸿
 * @desc: MVP模式Activity基类
 * @date Created on 2017/9/20 22:35
 * @github: https://github.com/lisuperhong
 */

public abstract class BaseMvpActivity<P extends BasePresenter> extends BaseActivity implements IMvpView {

    protected P presenter;

    @Override
    protected void onViewCreated() {
        super.onViewCreated();
        setPresenter();
        if (presenter != null) {
            presenter.attachView(this);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (presenter != null) {
            presenter.detachView();
        }
    }
}

File: CommonBusiness/src/main/java/com/company/commonbusiness/base/activity/BaseMvpActivity.kt
package com.company.commonbusiness.base.activity

import com.company.commonbusiness.base.mvp.IMvpPresenter

/**
 * @author 李昭鸿
 * @desc: MVP模式Activity基类
 * @date Created on 2017/9/20 22:35
 * @github: https://github.com/lisuperhong
 */

abstract class BaseMvpActivity<P : IMvpPresenter> : BaseActivity() {

    protected var presenter: P? = null

    override fun onDestroy() {
        super.onDestroy()
        if (presenter != null) {
            presenter!!.detachView() //释放资源
        }
        presenter = null
    }

    /**
     * 绑定Presenter
     */
    protected abstract fun setPresenter()
}
